MantidPlot/src/Process.cpp

```diff
@@ -1,85 +1,35 @@
-// clang-format off
-#include "MantidQtWidgets/Common/PythonThreading.h"
-// clang-format on
-
 #include "Process.h"
 
-#include <iostream>
-#include <stdexcept>
 #include <QCoreApplication>
+#include <QFileInfo>
+
+#if defined(Q_OS_LINUX)
+#include <QDir>
+#include <QFile>
+#elif defined(Q_OS_WIN)
+#define WIN32_LEAN_AND_MEAN
+#include <Psapi.h>
+#include "MantidQtWidgets/Common/QStringUtils.h"
+#elif defined(Q_OS_MAC)
+#include <libproc.h>
+#include <sys/sysctl.h>
+#endif
 
 namespace {
 
-class PyObjectNewReference {
-public:
-  explicit PyObjectNewReference(PyObject *object) : m_object(object) {}
-  ~PyObjectNewReference() { Py_XDECREF(m_object); }
-
-  PyObjectNewReference(const PyObjectNewReference &) = delete;
-  PyObjectNewReference &operator=(const PyObjectNewReference &) = delete;
-
-  PyObjectNewReference(PyObjectNewReference &&o) { *this = std::move(o); }
-
-  PyObjectNewReference &operator=(PyObjectNewReference &&other) {
-    this->m_object = other.m_object;
-    other.m_object = nullptr;
-    return *this;
-  }
-
-  inline PyObject *ptr() const { return m_object; }
-
-private:
-  PyObject *m_object;
-};
-
-/**
- * @brief Retrieve a named attribute
- * @param source The source object
- * @param name The name of the attribute
- * @return The attribute
- * @throws std::runtime_error if an error occurs retrieving the attribute
- */
-PyObjectNewReference attr(PyObject *source, const char *name) {
-  PyObjectNewReference attr(PyObject_GetAttrString(source, name));
-  if (attr.ptr()) {
-    return attr;
-  } else {
-    PyErr_Print();
-    throw std::runtime_error(std::string("Process: No attribute ") + name +
-                             " found");
-  }
-}
-
-/**
- * @brief Call a named function with an check for errors
- * @param source The source object
- * @param name The name of the attribute to call
- * @return The return value of the function
- * @throws std::runtime_error if an error occurs retrieving the attribute
- */
-PyObjectNewReference call(PyObject *source, const char *name) {
-  auto returnedAttr = attr(source, name);
-  auto result = PyObject_CallFunction(returnedAttr.ptr(), nullptr);
-  if (result)
-    return PyObjectNewReference(result);
-  else {
-    PyErr_Print();
-    throw std::runtime_error(std::string("Process: Error calling function ") +
-                             name);
-  }
+bool isOtherInstance(int64_t otherPID, QString otherExeName) {
+  static const int64_t ourPID(QCoreApplication::applicationPid());
+  if (otherPID == ourPID)
+    return false;
+
+  static const QString ourExeName(
+      QFileInfo(QCoreApplication::applicationFilePath()).fileName());
+  if (ourExeName == otherExeName)
```

Comment:
same here. Line 27 to 30 should just be `return ourExeName == otherExeName`

Reply:
Agreed here though. That would be far cleaner...